hooks/useLogin.ts

```diff
@@ -39,78 +32,79 @@ export function useLogin(config: LoginProps = {}) {
-          magic.user.getMetadata(),
-          getUserDetails(token),
-        ])
+  const reloadUser = useCallback(async () => {
+    if (!magic || !magic.user) {
+      return false
+    }
 
-        if ('error' in details || details instanceof LocalError) {
-          // this is a visible error and a breaking error
-          $setStatus(STATUS.FAILED)
-          $setError(details)
-          return
-        }
+    let token
+    try {
+      token = await magic.user.getIdToken()
+    } catch (error) {}
+
+    if (!token) {
+      if (redirect) {
+        // if redirect string is provided and we're not logged in, cya!
+        // if this is kept as a static Router.push, it _does not_ work
+        $router.push(redirect)
+        return false
+      }
 
-        if (details.statusCode && details.statusCode === 401) {
-          $setStatus(STATUS.NOT_FOUND)
-          $setError(new LocalError('No user found.', NO_MAGIC_USER))
-          return
-        }
+      // this is a visible error but not a breaking error
+      $setStatus(STATUS.NOT_FOUND)
+      $setError(new LocalError('No token available.', NO_MAGIC_TOKEN))
+      return false
+    }
+    try {
+      const [magicMd, details] = await Promise.all([
+        magic.user.getMetadata(),
+        getUserDetails(token),
+      ])
+
+      if ('error' in details || details instanceof LocalError) {
+        // this is a visible error and a breaking error
+        $setStatus(STATUS.FAILED)
+        $setError(details)
+        Promise.reject(details)
+        return false
+      }
 
-        $setStatus(STATUS.LOADED)
-        $setMetadata(details)
-        $setMagicMetadata(magicMd)
-      } catch (err) {
-        if (err.toString().indexOf('-32603') > -1) {
-          $setStatus(STATUS.NOT_FOUND)
-          return
-        }
+      if (details.statusCode && details.statusCode === 401) {
+        $setStatus(STATUS.NOT_FOUND)
+        $setError(new LocalError('No user found.', NO_MAGIC_USER))
+        return false
+      }
 
-        throw err
+      $setStatus(STATUS.LOADED)
+      $setMetadata(details)
+      $setMagicMetadata(magicMd)
+      return true
+    } catch (err) {
+      if (err.toString().indexOf('-32603') > -1) {
+        $setStatus(STATUS.NOT_FOUND)
+        return false
       }
-    }
 
-    if (!$metadata) {
-      checkLoggedIn().catch(e => {
-        if ($status === STATUS.LOADING) {
-          $setStatus(STATUS.FAILED)
+      throw err
+    }
+  }, [$router, redirect])
+  useEffect(() => {
+    const loadAndCheck = async () => {
+      if (!$metadata) {
+        try {
+          await reloadUser()
+        } catch (e) {
+          if ($status === STATUS.LOADING) {
+            $setStatus(STATUS.FAILED)
+          }
         }
-
-        // eslint-disable-next-line no-console
-        console.warn('general error!', e)
-      })
+      }
     }
-  }, [$metadata, $setMetadata, redirect, $status, $router])
-
+    loadAndCheck()
+  }, [$metadata, $status, reloadUser])
```

Comment:
I noticed this `$metadata` in the effect dependencies is causing an extra `/me` fetch to happen. It happens because `$setStatus` gets called before `$setMetadata`, I believe. But since we only want this effect to run once on page load, we could remove the `$metadata` check entirely, and also remove the `$status` check in the `catch` block to get status out of the dependencies as well. That should drop the `/me` calls back down to the same as on production.

Reply:
@dguenther You mean like:
```
  useEffect(() => {
    const loadAndCheck = async () => {
      try {
        await reloadUser()
      } catch (e) {
        $setStatus(STATUS.FAILED)
      }
    }
    loadAndCheck()
  }, [reloadUser])
```
?

Reply:
Yep, I believe so. That should do ~2 `/me` calls instead of 3.